Sources/Operators/Distinct.swift

```diff
@@ -0,0 +1,66 @@
+//
+//  Distinct.swift
+//  CombineExt
+//
+//  Created by Jasdev Singh on 21/03/2020.
+//  Copyright © 2020 Combine Community. All rights reserved.
+//
+
+import Combine
+
+public extension Publisher where Output: Hashable {
+    /// De-duplicates _all_
+    /// published value events, as opposed to pairwise with `Publisher.removeDuplicates`.
+    ///
+    /// - note: It’s important to note that this operator stores all emitted values
+    ///         in an in-memory `Set`. So, use this operator with caution, when handling publishers
+    ///         that emit a large number of unique value events.
+    ///
+    /// - returns: A publisher that consumes duplicate values across all previous emissions from upstream.
+    func distinct() -> Publishers.Filter<Self> {
+        var seen = Set<Output>()
+        return filter { incoming in seen.insert(incoming).inserted }
+    }
+}
+
+public extension Publisher where Output: Equatable {
+    /// `Publisher.distinct` de-duplicates _all_
+    /// published `Hashable`-conforming value events, as opposed to pairwise with `Publisher.removeDuplicates`.
+    ///
+    /// - note: It’s important to note that this operator stores all emitted values
+    ///         in an in-memory `Array`. So, use this operator with caution, when handling publishers
+    ///         that emit a large number of unique value events.
+    ///
+    /// - returns: A publisher that consumes duplicate values across all previous emissions from upstream.
+    func distinct() -> Publishers.Filter<Self> {
+        distinct(by: ==)
+    }
+}
+
+public extension Publisher {
+    /// `Publisher.distinct(by:)` de-duplicates _all_ published value events,
+    /// along the provided `by` predicate, as opposed to pairwise with `Publisher.removeDuplicates(by:)`.
+    ///
+    /// - parameter by: A predicate to use when determining uniqueness. `Publisher.distinct` will iterate
+    /// over all seen values applying each known unique value as the first argument to the predicate and the
+    /// incoming value event as the second, i.e. `by(see, next) -> Bool`. If this predicate is `true` for any
+    /// seen value, the next incoming value isn’t emitted downstream.
+    ///
+    /// - note: It’s important to note that this operator stores all emitted values
+    ///         in an in-memory `Array`. So, use this operator with caution, when handling publishers
+    ///         that emit a large number of unique value events (as per `by`).
+    ///
+    /// - returns: A publisher that consumes duplicate values across all previous emissions from upstream
+    ///            (signaled with `by`).
+    func distinct(by predicate: @escaping (Output, Output) -> Bool) -> Publishers.Filter<Self> {
```

Comment:
```suggestion
    func distinct(by comparator: @escaping (Output, Output) -> Bool) -> Publishers.Filter<Self> {
```

I think a predicate is when you're filtering for something, but here you're comparing elements